File: pinpoint/lib/model/address.dart
import 'package:uuid/uuid.dart';

class Address {
  final String id;
  final String streetAddress;
  final String city;
  final String state;
  final String postalCode;
  final String country;

  Address({
    required this.id,
    required this.streetAddress,
    required this.city,
    required this.state,
    required this.postalCode,
    required this.country,
  });

  factory Address.fromJson(Map<String, dynamic> json) => Address(
        id: json['id'] ?? const Uuid().v4(),
        streetAddress: json['streetAddress'],
        city: json['city'],
        state: json['state'],
        postalCode: json['postalCode'],
        country: json['country'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'streetAddress': streetAddress,
        'city': city,
        'state': state,
        'postalCode': postalCode,
        'country': country,
      };
}

File: pinpointB/src/main/java/com/nxquar/pinpoint/Model/Address.java
package com.nxquar.pinpoint.Model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    private String streetAddress;
    private String city;
    private String state;
    private String postalCode;
    private String country;
}